presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -882,7 +882,7 @@ protected Object visitFunctionCall(FunctionCall node, Object context)
             }
 
             // do not optimize non-deterministic functions
-            if (optimize && (!function.isDeterministic() || hasUnresolvedValue(argumentValues))) {
+            if (optimize && (!function.isDeterministic() || hasUnresolvedValue(argumentValues) || node.getName().toString().equals("grouping"))) {
```

Comment:
Just to confirm, you'd like me to remove this check and instead mark the function as non-deterministic because the definition of non-deterministic includes functions that should not be inlined?